apps/shared/src/components/pipeline-actions/PipelineActions.tsx

```
@@ -0,0 +1,127 @@
+ * The endpoint info comes from `taskStatus.notes[0]` when the server reports
+ * that a service is ready. This component extracts the button metadata and
+ * renders compact action buttons.
+ *
+ * Used on canvas source nodes and the status page header.
+ *
+ * Plain React + inline styles using --rr-* theme tokens. No MUI.
+ */
+
+import React, { ReactElement, useMemo, useState } from 'react';
+import type { CSSProperties } from 'react';
+import { Button } from 'shell';
```

ESLint

[error] 'shell' import is restricted from being used. Renamed: import from 'shell'.

(no-restricted-imports)